server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/webhook/WebhookPostTask.java

```diff
@@ -19,82 +19,64 @@
+    this.webHooks = webHooks;
   }
 
   @Override
   public void finished(ProjectAnalysis analysis) {
     Configuration config = configRepository.getConfiguration();
 
-    Iterable<String> webhookProps = Iterables.concat(
-      getWebhookProperties(config, WebhookProperties.GLOBAL_KEY),
-      getWebhookProperties(config, WebhookProperties.PROJECT_KEY));
-    if (!Iterables.isEmpty(webhookProps)) {
-      process(config, analysis, webhookProps);
-      deliveryStorage.purge(analysis.getProject().getUuid());
-    }
+    webHooks.sendProjectAnalysisUpdate(
+      config,
+      new WebHooks.Analysis(
+        analysis.getProject().getUuid(),
+        analysis.getAnalysis().map(org.sonar.api.ce.posttask.Analysis::getAnalysisUuid).orElse(null),
+        analysis.getCeTask().getId()),
+      () -> payloadFactory.create(convert(analysis)));
   }
 
-  private static List<String> getWebhookProperties(Configuration config, String propertyKey) {
-    String[] webhookIds = config.getStringArray(propertyKey);
-    return Arrays.stream(webhookIds)
-      .map(webhookId -> format("%s.%s", propertyKey, webhookId))
-      .limit(MAX_WEBHOOKS_PER_TYPE)
-      .collect(MoreCollectors.toList(webhookIds.length));
-  }
-
-  private void process(Configuration config, ProjectAnalysis analysis, Iterable<String> webhookProperties) {
-    WebhookPayload payload = payloadFactory.create(analysis);
-    for (String webhookProp : webhookProperties) {
-      String name = config.get(format("%s.%s", webhookProp, WebhookProperties.NAME_FIELD)).orElse(null);
-      String url = config.get(format("%s.%s", webhookProp, WebhookProperties.URL_FIELD)).orElse(null);
-      // as webhooks are defined as property sets, we can't ensure validity of fields on creation.
-      if (name != null && url != null) {
-        Webhook webhook = new Webhook(analysis.getProject().getUuid(), analysis.getCeTask().getId(), name, url);
-        WebhookDelivery delivery = caller.call(webhook, payload);
-        log(delivery);
-        deliveryStorage.persist(delivery);
-      }
-    }
-  }
+  private static org.sonar.server.webhook.ProjectAnalysis convert(ProjectAnalysis projectAnalysis) {
+    CeTask ceTask = new CeTask(projectAnalysis.getCeTask().getId(), CeTask.Status.valueOf(projectAnalysis.getCeTask().getStatus().name()));
+    Project project = new Project(projectAnalysis.getProject().getUuid(), projectAnalysis.getProject().getKey(), projectAnalysis.getProject().getName());
+    Analysis analysis = projectAnalysis.getAnalysis().map(a -> new Analysis(a.getAnalysisUuid(), a.getDate().getTime())).orElse(null);
+    Branch branch = projectAnalysis.getBranch().map(b -> new Branch(b.isMain(), b.getName().orElse(null), Branch.Type.valueOf(b.getType().name()))).orElse(null);
+    QualityGate qualityGate = Optional.ofNullable(projectAnalysis.getQualityGate())
+      .map(qg -> new QualityGate(
+        qg.getId(),
+        qg.getName(),
+        QualityGate.Status.valueOf(qg.getStatus().name()),
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "%s.%s" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)